docs/internals/release-process.txt

```diff
@@ -37,39 +35,84 @@ security purposes, please see :doc:`our security policies <security>`.
 
 .. glossary::
 
-  Major release
-    Major releases (A.B, A.B+1, etc.) will happen roughly every nine months --
-    see `release process`_, below for details. These releases will contain new
+  Feature release
+    Feature releases (A.B, A.B+1, etc.) will happen roughly every eight months
+    -- see `release process`_ for details. These releases will contain new
     features, improvements to existing features, and such.
 
-    .. _internal-release-deprecation-policy:
+  Patch release
+    Patch releases (A.B.C, A.B.C+1, etc.) will be issued as needed, to fix
+    bugs and/or security issues.
 
-    A major release may deprecate certain features from previous releases. If a
-    feature is deprecated in version ``A.B``, it will continue to work in versions
-    ``A.B`` and  ``A.B+1`` but raise warnings. It will be removed in version
-    ``A.B+2``.
+    These releases will be 100% compatible with the associated feature release,
+    unless this is impossible for security reasons or to prevent data loss.
+    So the answer to "should I upgrade to the latest patch release?" will always
+    be "yes."
+
+  Long-term support release
+    Certain feature releases will be designated as "Long-term support" (LTS)
+    releases. LTS releases will get security and data loss fixes applied for a
+    guaranteed period of time, typically three years.
 
-    So, for example, if we decided to start the deprecation of a function in
-    Django 1.7:
+    See `the download page`_ for the releases that have been designated for
+    long-term support.
 
-    * Django 1.7 will contain a backwards-compatible replica of the function which
-      will raise a ``RemovedInDjango19Warning``. This warning is silent by
-      default; you can turn on display of these warnings with the ``-Wd`` option
-      of Python.
+    .. _the download page: https://www.djangoproject.com/download/
 
-    * Django 1.8 will still contain the backwards-compatible replica. This
-      warning becomes *loud* by default, and will likely be quite annoying.
+Release cadence
+===============
 
-    * Django 1.9 will remove the feature outright.
+Starting with Django 3.0, version numbers will use a loose form of `semantic
+versioning <http://semver.org/>`_ such that each version following an LTS will
+bump to the next "dot zero" version. For example: 3.0, 3.1, 3.2 (LTS), 4.0,
+4.1, 4.2 (LTS), etc.
 
-  Minor release
-    Minor releases (A.B.C, etc.) will be issued as needed, often to fix security
-    issues.
+SemVer makes it easier to see at a glance how compatible releases are with each
+other. It also helps to anticipate when compatibility shims will be removed.
+It's not a pure form of SemVer as each feature release will continue to have a
+few documented backwards incompatibilities where a deprecation path isn't
+possible or not worth the cost. Also, deprecations started in the version
+preceding an LTS (X.1) will be dropped in a non-dot-zero release to accommodate
```

Comment:
Maybe add a parenthetical `(Y.1)` after `in a non-dot-zero release` to clarify which one?